### ISSUE TITLE
401 auth headers not passed

### ISSUE THREAD

**Issue by krazik** [sholladay/hapi-error-page#2]
If you combine this with hapi-basic-auth (use example setup),

the www-authenticate header from the boom.unauthenticated response is lost and user is never shown auth dialog.

**Comment by krazik** [comment]
how I fixed (forked locally)

```    const result = h.view('error', context).code(statusCode);
    if (response.output.headers) {
      for (const [key, value] of Object.entries(response.output.headers)) {
        result.header(key, value);
      }
    }
    return result;```

**Comment by sholladay** [comment]
Hmm yes, you are correct. I think I originally did that intentionally because rendering a view is a new response. I doubt we woukd want to copy over the `Content-Type` header, for example. Otherwise you could easily receive `Content-Type: application/json` along with an HTML error page.

However, I agree that there are probably some headers that should be forwarded, including `www-authenticate`, among others. That is an oversight. Happy to take a PR for that!

**Comment by sholladay** [comment]
@krazik I don't know if you are still using this plugin (or your fork of it), but I just published `1.0.0` with the fix for this issue!

https://github.com/sholladay/hapi-error-page/releases/tag/v1.0.0

I omitted the `if (response.output.headers) {` check from your code, because I couldn't quickly figure out any cases where that check would be falsey. Please let me know if you think it's important for some reason.

Also, I decided to ignore my previous comment and not filter any headers for now. I will do so if it becomes an actual problem for someone in the future, but after thinking about it some more, I don't think it will be in practice.

Thanks for your bug report and providing a solution!
